sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py

```diff
@@ -92,7 +97,7 @@ def set_authentication_policy(credential, kwargs):
     elif hasattr(credential, "get_token"):
         if not kwargs.get("authentication_policy"):
             if kwargs.get("region") and kwargs.get("resource_id"):
-                scope = kwargs.pop("audience", DEFAULT_ENTRA_ID_SCOPE).rstrip("/") + DEFAULT_SCOPE
+                scope = kwargs.pop("audience", DEFAULT_ENTRA_ID_SCOPE).rstrip("/").rstrip(DEFAULT_SCOPE) + DEFAULT_SCOPE
```

Comment:
Can you help me understand why we need to separate the pieces of the scope here? What are the expected values? In SDKs that do Entra ID authentication, we will just default it to "https://cognitiveservices.azure.com/.default" if not passed by the user. Is that something we can do here?

Reply:
Note: I will use AAD for Entra ID and I will use scopes for scopes and audiences.

When using AAD auth, the default value for the scopes is "https://cognitiveservices.azure.com/.default". All SDKs are expecting users to provide "https://cognitiveservices.azure.com" (in case of public prod, there are variants for sov. clouds and AGC). I added rstrip to handle "just in case" cases where users provides "https://cognitiveservices.azure.com/" or "https://cognitiveservices.azure.com/.default". (or any sov cloud, AGC variant).